spec/requests/api/v1/attendances_spec.rb

```diff
@@ -6,22 +6,42 @@
   let!(:logged_in_user) { create(:confirmed_user) }
   let!(:business) { create(:business, user: logged_in_user) }
   let!(:child) { create(:child, business: business) }
+
+  let!(:week_current_date) { Time.zone.now }
+  let!(:week_start_date) { week_current_date.at_beginning_of_week(:sunday) } # Sunday
+  let!(:week_end_date) { week_current_date.at_end_of_week(:saturday) } # Friday
+
+  let!(:two_weeks_ago_week_current_date) { week_current_date - 2.weeks }
+  let!(:two_weeks_ago_week_start_date) { week_start_date - 2.weeks }
+  let!(:two_weeks_ago_week_end_date) { week_end_date - 2.weeks }
+
   let!(:this_week_attendances) do
-    create_list(:attendance, 3, child_approval: child.child_approvals.first, check_in: Faker::Time.between(from: Time.current.at_beginning_of_week(:sunday), to: Time.current))
+    check_in_date = Faker::Time.between(from: week_start_date, to: week_end_date)
+
+    create_list(:attendance, 3, child_approval: child.child_approvals.first, check_in: check_in_date)
   end
+
   let!(:past_attendances) do
-    create_list(:attendance, 2, child_approval: child.child_approvals.first,
-                                check_in: Faker::Time.between(from: (Time.current - 2.weeks).at_beginning_of_week(:sunday), to: (Time.current - 2.weeks).at_end_of_week(:saturday)))
+    check_in_date = Faker::Time.between(from: two_weeks_ago_week_start_date, to: two_weeks_ago_week_end_date)
+
+    create_list(:attendance, 2, child_approval: child.child_approvals.first, check_in: check_in_date)
+  end
+
+  let!(:extra_attendances) do
+    create_list(:attendance, 3, check_in: Faker::Time.between(from: week_start_date, to: week_current_date))
   end
-  let!(:extra_attendances) { create_list(:attendance, 3, check_in: Faker::Time.between(from: Time.current.at_beginning_of_week(:sunday), to: Time.current)) }
 
   describe 'GET /api/v1/attendances' do
     include_context 'correct api version header'
 
-    before { sign_in logged_in_user }
+    before do
+      sign_in logged_in_user
+      allow(logged_in_user).to receive(:timezone).and_return('UTC')
```

Comment:
Part of the flakiness was due to the way the app changes the Time.zone to the logged-in user's timezone. So with RSpec, timezone change happens and make time and date inconsistent with the examples

Reply:
I changed the test environment configuration to set the timezone to UTC - I think if we force the *user* to UTC we might miss some behavior that's due to a user logging in from (say) the Central time zone.  We're explicitly storing attendances in UTC but we want to make sure that if someone logs in at 8am Central, they're seeing the attendances we expect.

Can you rebase from develop, which has the configuration change, and see if you can remove this particular `allow` statement and still get the test to pass?